### PR TITLE
Feat/slowquery v2

### PR DESCRIPTION
**Pull request by shhdgit**
image
image
image

### CODE REVIEW

File: ui/packages/tidb-dashboard-lib/tsconfig.json [hunk 1]
```diff
@@ -11,5 +11,5 @@
       "@lib/*": ["src/*"]
     }
   },
-  "include": ["src"]
+  "include": ["src", "node_modules/@diag-ui"]
```
Comment (baurine) [inline ui/packages/tidb-dashboard-lib/tsconfig.json:14]: Why we need to include this package?

**Comment by baurine** [comment]
do we have any spec doc?

File: ui/packages/tidb-dashboard-lib/tsconfig.json (hunk 1 above)
Comment (shhdgit) [reply ui/packages/tidb-dashboard-lib/tsconfig.json:14]: It's for dev. No longer needed.

File: ui/packages/tidb-dashboard-for-clinic-cloud/src/apps/SlowQuery/context.ts [hunk 2]
```diff
@@ -80,6 +80,28 @@ class DataSource implements ISlowQueryDataSource {
...
+      )
+  }
+
+  promqlQueryRange(query: string, start: number, end: number, step: string) {
+    return client
+      .getAxiosInstance()
+      .get(
+        `/slow_query/vm_query_range?query=${query}&start=${start}&end=${end}&step=${step}`
+      )
+  }
```
[the remark this answers is not in the document]
Comment (shhdgit) [reply ui/packages/tidb-dashboard-for-clinic-cloud/src/apps/SlowQuery/context.ts:104]: under the `/ngm/api/v1` namespace.

File: ui/packages/tidb-dashboard-for-clinic-cloud/src/apps/SlowQuery/context.ts [hunk 3]
```diff
@@ -80,6 +80,30 @@ class DataSource implements ISlowQueryDataSource {
...
+
+  promqlQueryRange(query: string, start: number, end: number, step: string) {
+    return client
+      .getAxiosInstance()
+      .get(
+        `/slow_query/vm_query_range?query=${query}&start=${start}&end=${end}&step=${step}`
+      )
+      .then((res) => res.data)
+  }
```
Comment (baurine) [inline ui/packages/tidb-dashboard-for-clinic-cloud/src/apps/SlowQuery/context.ts:106]: I'm not sure, whether can we define these APIs in the tidb-dashboard backend so we can use the generated API methods.

**Review by YiniXu9506 — APPROVED** [review]

File: ui/packages/tidb-dashboard-for-clinic-cloud/src/apps/SlowQuery/context.ts (hunk 3 above)
Comment (shhdgit) [reply ui/packages/tidb-dashboard-for-clinic-cloud/src/apps/SlowQuery/context.ts:106]: It looks like a special api from ngm.

File: cmd/tidb-dashboard/main.go [hunk 4]
```diff
@@ -1,4 +1,4 @@
-// Copyright 2022 PingCAP, Inc. Licensed under Apache-2.0.
+// Copyright 2023 PingCAP, Inc. Licensed under Apache-2.0.
```
Comment (baurine) [inline cmd/tidb-dashboard/main.go:1]: I think we should handle this change in an individual PR, instead of mixing with another feature PR.

**Pull request merged**